lib/state/state.js

```diff
@@ -517,36 +481,63 @@ State.prototype.respond = function (response, isJson) {
 	this.response = isJson ? response : JSON.stringify(response);
 };
 
+/**
+ * Distribute all events currently stacked on
+ * this state object.
+ *
+ * If you wish to also distribute archivist mutations stacked
+ * on this state, please have a look at the `distribute` method
+ * instead.
+ *
+ * @param {Function} cb callback function
+ */
+State.prototype.distributeEvents = function (cb) {
+	this.otherEvents[this.actorId] = this.myEvents.slice();
+	this.myEvents.length = 0;
 
-function closeWithError(state, cb) {
-	cb(null, {
-		errorCode: state.errorCode,
-		myEvents: state.myEvents.length ? state.myEvents : undefined
+	this.emitEvents(false, cb);
+};
+
+/**
+ * Distribute both events and archivist operations
+ * currently stacked on this state object.
+ *
+ * @param {Function} cb callback function
+ */
+State.prototype.distribute = function (cb) {
+	this.archivist.distribute((error) => {
+		if (error) {
+			return cb(error);
+		}
+
+		this.distributeEvents(cb);
 	});
-}
+};
 
 
-function closeWithSuccess(state, cb) {
-	state.archivist.distribute(function (error) {
-		// archivist errors can only come from the operations before writing anything to a vault,
-		// (beforeDistribute hooks, operation validation, etc), so it is still safe to bail out and
-		// report an error.
+function destroy(state, response, cb) {
+	// cleanup
+	state.destroy();
 
-		if (error) {
-			return closeWithError(state, cb);
-		}
+	// return the response
+	if (cb) {
+		cb(null, response);
+	}
+}
```

Comment:
The `destroy` function serves little (actually.. it serves no) purpose. I would just inline this in `respond`.